src/metabase/pulse/render/body.clj

```diff
@@ -548,29 +548,31 @@
 
   overlap = (/ overlap-width max-width) = (/ 35 59) = 0.59"
   [col-a col-b]
-  (let [[min-a min-b]    (map #(get-in % [:fingerprint :type :type/Number :min]) [col-a col-b])
-        [max-a max-b]    (map #(get-in % [:fingerprint :type :type/Number :max]) [col-a col-b])
-        non-overlapping? (or (and (< max-a min-b) (< max-a max-b))
-                             (and (> min-a min-b) (> min-a max-b)))]
-   (if non-overlapping?
-     0
-     (let [[a b c d]     (sort [min-a min-b max-a max-b])
-           max-width     (- d a)
-           overlap-width (- c b)]
-       (/ overlap-width max-width)))))
+  (let [[min-a min-b]    (map #(get-in % [:fingerprint :type :type/Number :min] 0) [col-a col-b])
+        [max-a max-b]    (map #(get-in % [:fingerprint :type :type/Number :max] 0) [col-a col-b])
+        valid-ranges?    (and min-a min-b max-a max-b)
+        non-overlapping? (or (and (< max-a min-b) (< max-a max-b) valid-ranges?)
+                             (and (> min-a min-b) (> min-a max-b) valid-ranges?))]
+    (if
+      non-overlapping?
```

Comment:
Maybe I'm reading this wrong, but two functional issues:

1. I think it's possible for one of the min/max variables to be `0` because it's `0`, and that doesn't mean the range is invalid.

2. If the range is invalid (as you define it), then `non-overlapping?` will be `false`, which means that regardless of other factors you'll treat the case as an overlapping range. I suspect that you want invalid ranges to trigger the *non* overlapping case.

For #2, we're in a murky territory with double negatives, so it might be clearer if you renamed `non-overlapping?` to `overlapping?` and flipped the logic around using [DeMorgan's](https://en.wikipedia.org/wiki/De_Morgan%27s_laws).

Reply:
Yea, good call. I'll fix it up